libtextsecure/http-resources.js

```diff
@@ -68,7 +68,19 @@
         setTimeout(() => { pollServer(callBack); }, 5000);
         return;
       }
-      result.messages.forEach(async message => {
+      const incomingHashes = result.messages.map(m => m.hash);
+      const dupHashes = await window.Signal.Data.getSeenMessagesByHashList(incomingHashes);
+      if (incomingHashes.length === dupHashes.length) {
+        setTimeout(() => { pollServer(callBack); }, 5000);
+        return;
+      }
+      const NewMessages = result.messages.filter(m => !dupHashes.includes(m.hash));
+      const NewHashes = NewMessages.map(m => ({
```

Comment:
de-capitalise those two variables?

Reply:
Maybe for readability, instead of doing everything here, move it to a new function:
```js
const newMessages = this.filterExistingMessages(result.messages);
newMessages.forEach(async message => {
...
```